src/main/pages/che-7/administration-guide/proc_configuring-prod-short-to-use-external-keycloak.adoc

```diff
@@ -0,0 +1,77 @@
+
+.Prerequisites
+
+ifeval::["{project-context}" == "che"]
+* This procedure is only applicable to {prod-short} installations done using the {prod-short} Operator. When using the `{prod-cli}` management tool and the `helm` installation mode to install {prod-short}, no supported method is available to use an external {identity-provider} instance.  
+endif::[]
+
+*  In the administration console of the {identity-provider} external installation, define a link:https://www.keycloak.org/docs/latest/server_admin/#_create-realm[realm] that will contain the users intended to connect to {prod-short}. The following steps use the example `realm-for-users` realm name.
++
+image::keycloak/external_keycloak_realm.png[link="{imagesdir}/keycloak/external_keycloak_realm.png]
+
+
+* In this `realm`, define an link:https://www.keycloak.org/docs/latest/server_admin/#oidc-clients[OIDC client] that {prod-short} will use to authenticate the users. The following steps use the example `public-client` name. Here is an example of such a client with the correct settings:
++
+image::keycloak/external_keycloak_public_client.png[link="{imagesdir}/keycloak/external_keycloak_public_client.png]
+
+
+[CAUTION]
+====
+* {prod-short} only supports `public` OIDC clients. Therefore, selecting the `openid-connect` `Client Protocol` option and the `public` Access Type option is highly recommended.
+
+* The list of `Valid Redirect URIs` must contain at least 2 URIs related to the {prod-short} server, one using the `http` protocol and the other `https`. These URIs must contain the base URL of the {prod-short} server, followed by `/*` wildcards.
+
+* The list of `Web Origins` must contain at least 2 URIs related to the Che server, one using the `http` protocol and the other `https`. These URIs must contain the base URL of the {prod-short} server, without any path after the host.
++
+The number of URIs depends on the number of installed product tools.
+====
+
+* If {prod-short} is installed
+ifeval::["{project-context}" == "che"]
+on OpenShift
+endif::[]
+and uses the default OpenShift OAuth support, user authentication relies on the integration of {identity-provider} with OpenShift OAuth. This allows users to log in to {prod-short} with their OpenShift login and have their workspaces created under personal OpenShift namespaces. 
++
+This requires setting up an OpenShift identity provider inside {identity-provider}. When using an external {identity-provider}, set up the identity provider manually. For instructions, seethe appropriate {identity-provider} documentations for either link:https://www.keycloak.org/docs/latest/server_admin/#openshift-3[OpenShift 3] or link:https://www.keycloak.org/docs/latest/server_admin/#openshift-4[OpenShift 4].
```

Comment:
typo: `seethe ` :D Unless you mean you're super duper angry :P

Reply:
should we link to the SSO docs instead of the Keycloak ones?

Reply:
well, it will look exactly the same, but sure.

Reply:
Yes, that's the only thing I've not done in this PR: we should use attributes here to manage different URL for upstream and downstream.
@MichalMaler Do you plan to do it or should I do it myself ?

Reply:
Will do it.

Reply:
Here is the Downstream MR that will add the CRW identity provider links to the downstream docs
https://gitlab.cee.redhat.com/red-hat-developers-documentation/red-hat-devtools/-/merge_requests/715